### ISSUE TITLE
Difference between Embeddings

### ISSUE THREAD

**Issue by CDitzel** [lucidrains/vit-pytorch#59]
Hi, me again. Skimming through your repositories, I am wondering, what is the difference between using embeddings like

nn.Embedding(num_tokens, dim)

vs. 

nn.Linear(num_tokens, dim)

vs. 

 nn.Parameter(torch.randn(num_tokens, dim))

I have seen all three in code of yours. All of them are trainable, so is it just a matter of style?
